lib/ansible/modules/system/journald.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2018, Susant Sahani <susant@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['stableinterface'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: journald
+short_description: Automates journald conf files.
+description:
+    - Allows you to generate journald configuration files.
+version_added: "2.8"
+options:
+    conf_path:
+        description:
+            - Specifies the path where to write the configuration files.
+        default: "/etc/systemd/journald.conf"
+        choices: [ "/etc/systemd/journald.conf", "/etc/systemd/journald.conf.d",
+                   "/usr/lib/systemd/journald.conf.d", "/run/systemd/journald.conf.d" ]
+    file_name:
+        description:
+            - This configuration file name where the configurations will be written. Note that file name will be
+              automatically have the extension .conf. When path is /etc/systemd/journald.conf then the file_name
+              will be ignored.
+    storage:
+        description:
+            - Specifies where to store journal data.
+        choices: [ "volatile", "persistent", "auto", "none" ]
+    compress:
+        description:
+            - Whether to comppress the data.
+        choices: [ "yes", "no" ]
+    ratelimit_interval:
+        description:
+            - Configures the rate limiting that is applied to all messages generated on the system.
+              May be specified in following units "s", "min", "h", "ms", "us".
+    ratelimit_burst:
+        description:
+            - Configures how many messages allowed in ratelimit_interval.
+    forward_to_syslog:
+        description:
+            - Configures whether log messages received by the journal daemon shall be forwarded to a syslog daemon.
+        choices: [ "yes", "no" ]
+    forward_to_kmsg:
+        description:
+            - Configures whether forwarded the kernel log buffer (kmsg).
+        choices: [ "yes", "no" ]
+    forward_to_console:
+        description:
+            - Configures whether forwarded console.
+        choices: [ "yes", "no" ]
+    forward_to_wall:
+        description:
+            - Configures whether forwarded as the wall messages to all logged-in users.
+        choices: [ "yes", "no" ]
+    state:
+        description:
+            - Create or removes the configration file.
+        choices: [ create, remove ]
```

Comment:
these are not 'states', they are actions

Reply:
Thanks updated.